office-365-feed/operations.py

```diff
@@ -0,0 +1,275 @@
+        raise ConnectorError(str(Err))
+
+
+def make_api_call(params):
+    # Retrieves all entries from the feed.
+    try:
+        result = []
+        for feed_obj in params.get('urls_list'):
+            feed_url = feed_obj.get('feedURL', '')
+            try:
+                response = requests.get(
+                    url=feed_url,
+                    verify=params.get('verify_ssl')
+                )
+                response.raise_for_status()
+                data = response.json()
+                if "latest" in data:
+                    result.append(data)
+                else:
+                    indicators = [i for i in data if 'ips' in i or 'urls' in i]  # filter empty entries
+                    result.extend(indicators)
+            except requests.exceptions.SSLError:
+                logger.error('An SSL error occurred.')
+                raise ConnectorError('An SSL error occurred.')
+            except requests.exceptions.ConnectionError:
+                logger.error('A connection error occurred.')
+                raise ConnectorError('A connection error occurred.')
+            except Exception as err:
+                logger.error(err)
+                raise ConnectorError(err)
+        return result
+    except Exception as Err:
+        logger.error(str(Err))
+        raise ConnectorError(str(Err))
+
+
+def get_latest_version(config):
+    try:
+        client_request_id = str(uuid.uuid4())
+        regions_list = convert_str_list(config.get('regions'))
+        params = {"urls_list": build_urls_dict(regions_list, services_list=["All"], unique_id=client_request_id,
+                                               endpoint="version"),
+                  "verify_ssl": config.get("verify_ssl")}
+
+        resp = make_api_call(params)
+        return resp
+    except Exception as Err:
+        logger.error(str(Err))
+        raise ConnectorError(str(Err))
+
+
+def fetch_last_updated_time(config, last_pull_time):
+    region = config.get('regions')
+    latest_version_list = get_latest_version(config)
+    for version in latest_version_list:
+        if version["instance"] == region:
+            latest_time = version["latest"]
+            try:
+                old_version_ts = datetime.datetime.strptime(str(last_pull_time), "%Y-%m-%dT%H:%M:%SZ").timestamp()
+            except:
+                old_version_ts = last_pull_time
+            latest_version_ts = int(datetime.datetime.strptime(latest_time, "%Y%m%d%H").timestamp())
+            if old_version_ts < latest_version_ts:
+                return True, latest_version_list
+    return False, None
+
+
+def handle_dedup(iterable):
+    # Remove duplicate entries from result
+    try:
+        seen = set()
+        result = []
+
+        for dic in iterable:
+            key = (dic['value'])
+            if key in seen:
+                continue
+            result.append(dic)
+            seen.add(key)
+        return result
+
+    except Exception as Err:
```

Comment:
try except where we raise the error back is not useful. just increases overall execution time. we can remove try catch.